modules/evm/src/tests.rs

```diff
@@ -513,11 +587,21 @@ fn contract_should_deploy_contracts() {
 	new_test_ext().execute_with(|| {
 		let result = <Test as Trait>::Runner::create(alice(), contract.clone(), 0, 1000000000).unwrap();
 		assert_eq!(result.exit_reason, ExitReason::Succeed(ExitSucceed::Returned));
+		assert_eq!(
+			balance(alice()),
+			INITIAL_BALANCE - <Test as Trait>::ContractExistentialDeposit::get()
+		);
+		assert_eq!(
+			reserved_balance(result.address),
+			<Test as Trait>::ContractExistentialDeposit::get()
+		);
 
 		// Factory.createContract
+		// need factory contract pay for the ContractExistentialDeposit. But factory not
```

Comment:
@xlc @zjb0807 I don't see why factory should be payable. Alice is willing to create the contract so she should pay the deposit

Reply:
non payable method means caller cannot send ACA and call the method. and we should not charge extra token from caller.